x-pack/plugins/observability_solution/observability_ai_assistant/public/components/insight/insight.tsx

```diff
@@ -290,9 +292,20 @@ export function Insight({
   };
 
   const {
-    services: { http },
+    services: {
+      http,
+      plugins: {
+        start: { licensing },
+      },
+    },
   } = useKibana();
 
+  const license = useObservable<ILicense | null>(licensing.license$);
+  const hasEnterpriseLicense = license?.hasAtLeast('enterprise');
+  if (isEmpty(connectors.connectors) || !hasEnterpriseLicense) {
```

Comment:
should we do this in the consumer component? this might create some weird margins if the consumer component renders it in a flex item

Reply:
I initially added the checks outside the component together with the `isEnabled` flag. However, that made the component async, so the consumer would have to `await` the component.

https://github.com/elastic/kibana/blob/main/x-pack/plugins/observability_solution/observability_ai_assistant/public/plugin.tsx#L100

What alternatives do you see?

Reply:
Ah... You mean, every consumer should be responsible for doing the check?

Reply:
I could add a helper to make that easier though.

Reply:
Hmmm I see your point. We can add an `useAvailability()` hook that takes care of the licensing. we already have a `useGenAIConnectors()` so maybe we can combine those two. Or we can leave it for now, do you have a screenshot of how this looks in the places where we use contextual insights?

Reply:
I check a few different ones, and even when wrapped in flex they still look fine to me.

## Log rate analysis

<img width="1000" alt="image" src="https://github.com/elastic/kibana/assets/209966/bb46d8f7-e5b0-477b-8b3c-6d3a6d5f87b9">

---

<img width="1000" alt="image" src="https://github.com/elastic/kibana/assets/209966/6373013d-588f-483e-80ef-141f48a624c0">

## APM error

![image](https://github.com/elastic/kibana/assets/209966/4a81f77b-5968-44dc-ade4-4dc5d980efe2)

---

![image](https://github.com/elastic/kibana/assets/209966/6baff36a-7f1b-4f0d-b40f-dd79beb41332)